src/compiler/debug.ts

```diff
@@ -356,6 +356,15 @@ export namespace Debug {
         }
     }
 
+    /**
+     * Asserts the symbol is defined and is of the right kind (originating in TSC or sample DTE depending o the test that is currently being run)
+     * The default implementation just asserts the symbol is not null
+     * In tests it is overridden to ensure we don't accidentally use TSC symbols in DTE
+     */
+    // eslint-disable-next-line prefer-const
+    export let assertSymbolValid = (symbol: Symbol) => {
+        assert(symbol, "Symbol not defined");
+    };

```

Comment:
I feel weird about this backdoor for testing. Is there any way to test the same thing without it? Was this just helpful once during testing and doesn't need to stick around? Reading https://github.com/microsoft/TypeScript/commit/6803a00c5c25093b4b420ddb82b80234dbe91461 it's only called from with isolated declaration emit; is it the case that the injected check could just always be present?

Reply:
The problem is that both `transpileDeclaration` and TSC emit use the `symbol` property on the node. TSC will put a full symbol in it, while `transpileDeclaration` will put a minimal symbol that has enough information to make declaration emit work.

During tests we reuse the ASTs between running the same test. As long as both emit methods go though their binding phase and set the symbol, this is not a problem. However a bug in the binder used by `transpileDeclaration` meant that a node had a TSC symbol instead of a `transpileDeclaration` one. The problem is that everything worked fine, while calling `transpileDeclaration`, the declaration transform happily used  the TSC symbol (which it should be able to do, declaration emit with `--isolatedDeclarations` in TSC will not use the binder in `transpileDeclaration`). 

This masked the bug, until I tried to actually use `transpileDeclaration` in context where the AST was not a hand me down from regular TSC, and so the symbol was missing.

The `assertSymbolValid ` will assert the symbol is not null while in TSC, and in tests that call `transpileDeclaration`  it will assert that teh symbol is not missing AND that it is a minimal symbol that comes from the `transpileDeclaration` declaration binder, ensuring that `transpileDeclaration` doesn't accidentally use TSC symbols. 

The reason this needs to be injected is that the same code paths are used both in TSC and in `transpileDeclaration` always asserting it's a TSC symbol or a `transpileDeclaration` minimal symbol is not an option.

Reply:
> (which it should be able to do, declaration emit with --isolatedDeclarations in TSC will not use the binder in transpileDeclaration).

Hm, I guess I thought that it would?

I'm still not happy with this backdoor, but I also can't think of a better way.